api/gateway-operator/v2beta1/gatewayconfiguration_types.go

```diff
@@ -142,6 +142,8 @@ type GatewayConfigDataPlaneServices struct {
 	// Services) or only internally (e.g. ClusterIP), and inject any additional
 	// annotations you need on the service (for instance, if you need to
 	// influence a cloud provider LoadBalancer configuration).
+	// The `port` in the ports of ingress service must be the same as the port
+	// in one of the Gateway's listeners.
```

Comment:
Is it only 1  port that has to match? I believe all should match.

I think it'd be better if we're more explicit here so that users know how to specify these options.